test/nodetool/test_flush.py

```diff
@@ -8,7 +8,22 @@
 import utils
 
 
-def test_flush(nodetool):
+def test_flush_all_tables(nodetool, scylla_only):
+    nodetool("flush", expected_requests=[
+        expected_request("POST", "/storage_service/flush")
+    ])
```

Comment:
Something very strange is happening here, and I'd like @denesb 's opinion.
You decided to change the behavior of "nodetool flush" (to call some new Scylla-only API), but decided to do this only in the new nodetool command that nobody uses yet - and not in the Java nodetool. And these tests verify that this is the case. But why? Why should the C++ nodetool behave differently from the Java one? The whole point of this test file is to verify that it doesn't. At least not yet (I'm sure that in the future we'll stop adding features to the old nodetool, but have we reached that state?)

Reply:
@nyh it doesn't change the "black-box" behavior from the user's perspective.
It achieves the same goal for bare `nodetool flush` in either `nodetool` implementations.
It is a mistake to have scylla_nodetool just mimic the exact api calls that happen to happen in the json nodetool implementation over scylla-jmx, as it inhibits progress.
If the semantics of the operation are indistinguishable I see no reason not to evolve the underlying implementation.

Reply:
I agree, as long as there are no semantical differences, the two implementations can diverge.
For the most part, there is no reason to diverge, but when there are more optimal set of API endpoints available, there is no reason to not use it.
I suspect we will see more and more changes like this, and slowly, even semantic differences will creep in. The reason is that updating the Java nodetool is such a chore, that everybody avoids it as much as they can.